### ISSUE TITLE
More research (Discord)

### ISSUE THREAD

**Issue by alextusinean** [Ygnecz/MTLic#5]
Hello, if anyone is interested and wants to research more of RouterOS, I've created a Discord server that can be joined below.
Especially:
- modifying ROS by changing the public key it uses for NPK file verification so that they can be signed with a private key other than MikroTik's (and automatically being able to modify the any package, including the system one, or creating new ones);
- modifying the system package's /nova/bin/keyman (which seems to be the one that verifies licenses) by changing the public key it uses for license verification so that they can be signed with a private key other than MikroTik's.

https://discord.gg/pykP2QjCJ3

**Comment by alextusinean** [comment]
I accidentally deleted a letter in the invite link. I edited the previous comment and it should work now.

**Comment by kamiliahs** [comment]
Hello, have you created a generator with this code? I don't know much about reverse engineering.

**Comment by alextusinean** [comment]
https://github.com/alextusinean/roskg

**Comment by caiot5** [comment]
Looks like, discord link expired. I was hoping to find where is the pubkey that one need to change in order to use roskg.

**Comment by alextusinean** [comment]
It's not expired as I can see

**Comment by mrkenz9595** [comment]
@alextusinean https://github.com/alextusinean/roskg can't access ?

**Comment by Ygnecz** [comment]
> Hello, if anyone is interested and wants to research more of RouterOS, I've created a Discord server that can be joined below. Especially:
> 
>     * modifying ROS by changing the public key it uses for NPK file verification so that they can be signed with a private key other than MikroTik's (and automatically being able to modify the any package, including the system one, or creating new ones);
> 
>     * modifying the system package's /nova/bin/keyman (which seems to be the one that verifies licenses) by changing the public key it uses for license verification so that they can be signed with a private key other than MikroTik's.
> 
> 
> https://discord.gg/pykP2QjCJ3

Thank you 👍

**Comment by Ygnecz** [comment]
> https://github.com/alextusinean/roskg

@alextusinean repository removed?

**Comment by thedix** [comment]
> repository removed?

@Ygnecz Yes, it is removed, but forks are left [intact](https://github.com/Phuong39/roskg). Also you should take a look at this [project](https://github.com/elseif/MikroTikPatch) (and [fork](https://github.com/loskiq/MikroTikPatch)) based on your and Alex's work. Thank you!

**Comment by parsitc** [comment]
Hello,

According to the above project, I am looking for a tool that can generate a license code for MikroTik router boards. However, I actually do not have any knowledge of programming. Those who have the ability to do this, please contact directly via email. parsitc@gmail.com
